### ISSUE TITLE
Is bce_w_logits equivalent to the pytorch BCEWithLogitsLoss?

### ISSUE THREAD

**Issue by Saipraneet** [ddehueck/jax-loss#4]
https://github.com/ddehueck/jax-loss/blob/410d1f59539a836c860ff37000360e344f9d4a85/jax_loss/classification.py#L4

The comment seems to indicate that this is indeed true, but I wanted to confirm.

**Comment by ddehueck** [comment]
Yes, it should be. Let me know if you find any differing results though.

**Comment by Saipraneet** [comment]
Thank you for such a prompt reply! Would it be possible to add a source for the formula used since it is not obvious?

**Comment by ddehueck** [comment]
No problem, see this discussion on the pytorch repo were I ended up pulling most of the code from for this. https://github.com/pytorch/pytorch/pull/1792

There are some mathematical tricks and computational things needed for a stable implementation.

**Comment by Saipraneet** [comment]
From what I understand, the code in the discussion you linked uses this:

`    neg_abs = - input.abs()`
`    loss = input.clamp(min=0) - input * target + (1 + neg_abs.exp()).log()`

This seems to be slightly different from the below line
https://github.com/ddehueck/jax-loss/blob/410d1f59539a836c860ff37000360e344f9d4a85/jax_loss/classification.py#L16

**Comment by ddehueck** [comment]
Later in the discussion, there seem to be a couple of bugs with that implementation then if you follow it to the eventual PR you'll end up at https://github.com/pytorch/pytorch/pull/2221/files

**Comment by Saipraneet** [comment]
Perfect, thank you for your help!

**Comment by ddehueck** [comment]
No problem!
